clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -609,14 +686,15 @@ public synchronized void handleCompletedBatch(ProducerBatch batch, ProduceRespon
     }
 
     public synchronized void transitionToUninitialized(RuntimeException exception) {
-        transitionTo(State.UNINITIALIZED);
+        transitionTo(State.UNINITIALIZED, exception, InvalidStateDetectionStrategy.BACKGROUND);
```

Comment:
I have this last open question, other than that, LGTM

Reply:
I think not, it will be ignored because of the UNINITIALIZED state, snippet from transitionTo:
```
else if (target == State.FATAL_ERROR || target == State.ABORTABLE_ERROR) {
            if (error == null)
                throw new IllegalArgumentException("Cannot transition to " + target + " with a null exception");
            lastError = error;
}
```
If it's ignored, do we need to log it somewhere?